internal/cmd/skupper/link/nonkube/link_generate.go

```diff
@@ -4,30 +4,113 @@ Copyright © 2024 Skupper Team <skupper@googlegroups.com>
 package nonkube
 
 import (
+	"errors"
 	"fmt"
-
 	"github.com/skupperproject/skupper/internal/cmd/skupper/common"
+	"github.com/skupperproject/skupper/internal/nonkube/client/fs"
+	nonkubecommon "github.com/skupperproject/skupper/internal/nonkube/common"
+	"github.com/skupperproject/skupper/pkg/nonkube/api"
 	"github.com/spf13/cobra"
+	"os"
+	"strings"
 )
 
 type CmdLinkGenerate struct {
-	CobraCmd  *cobra.Command
-	Namespace string
-	siteName  string
-	Flags     *common.CommandLinkGenerateFlags
+	CobraCmd     *cobra.Command
+	Namespace    string
+	siteName     string
+	tokenHandler *fs.TokenHandler
+	siteState    *api.SiteState
+	Flags        *common.CommandLinkGenerateFlags
+	linkName     string
+	endpointHost string
 }
 
 func NewCmdLinkGenerate() *CmdLinkGenerate {
 	return &CmdLinkGenerate{}
+
 }
 
 func (cmd *CmdLinkGenerate) NewClient(cobraCommand *cobra.Command, args []string) {
-	//TODO
+	if cmd.CobraCmd != nil && cmd.CobraCmd.Flag(common.FlagNameNamespace) != nil && cmd.CobraCmd.Flag(common.FlagNameNamespace).Value.String() != "" {
+		cmd.Namespace = cmd.CobraCmd.Flag(common.FlagNameNamespace).Value.String()
+	}
+
+	cmd.tokenHandler = fs.NewTokenHandler(cmd.Namespace)
 }
 
-func (cmd *CmdLinkGenerate) ValidateInput(args []string) error { return nil }
-func (cmd *CmdLinkGenerate) InputToOptions()                   {}
+func (cmd *CmdLinkGenerate) ValidateInput(args []string) error {
+
+	var validationErrors []error
+
+	if len(args) > 0 {
+		validationErrors = append(validationErrors, fmt.Errorf("arguments are not allowed in this command"))
+	}
+
+	pathProvider := fs.PathProvider{Namespace: cmd.Namespace}
+	siteStateLoader := &nonkubecommon.FileSystemSiteStateLoader{
+		Path: pathProvider.GetRuntimeNamespace(),
+	}
+
+	siteState, err := siteStateLoader.Load()
+	if err != nil {
+		validationErrors = append(validationErrors, fmt.Errorf("there is no active site in this namespace"))
+	} else {
+
+		hasRouterAccess := false
+		if siteState.RouterAccesses != nil && len(siteState.RouterAccesses) > 0 {
+			for _, access := range siteState.RouterAccesses {
+				if strings.HasPrefix(access.Name, "router-access") {
```

Comment:
The resource names are not meant to be prefixed as "router-access". The file names were prefixed
as such some time ago. But now, if you rebase on top of main, the resources are serialized through file
names prefixed with the kind, which in this case would be `RouterAccess`.
But just the filename, not the resource name.

I believe that an accurate solution would be to add a method into the SiteState type that iterates through
the `RouterAccesses`, and for each RouterAccess, it then iterates through their roles and returning true
if a RouterAccess is defined with the roles: `inter-router` or `edge`.

Similar to the existing method: `HasRouterAccess`.

Reply:
Got it! thanks